accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,126 @@
+
+When a component ships both in a NuGet package and a framework, the developer has a consistent experience with the source generator as they do with the library itself.  If the package referenced is newer then the framework the source generator from the package will be used as is already the case for the library in that package.  Similarly if the framework is newer than the NuGet package the framework's source generator will be used as is already the case for the library in the framework.
+
+A developer can recieve updates to a source generator through normal channels which they already obtain updates today.
+
+A developer can update their version of Visual Studio or Dotnet SDK and have confidence that new versions of the toolchain won't impact their applications which target older frameworks, outside of known and published servicing updates.
+
+A developer has confidence that their library using source generators will continue to work in the same way a library without source generators works on future versions of .NET following the same compatibility rules.
+
+## Requirements
+
+### Goals
+
+- Enable consumption of source generators for components in shared frameworks
+- Address conflict resolution between source generators in shared frameworks and nuget packages
+- Define servicing process for shared framework source generators
+- Establish guidelines building and versioning shared framework source generators
+- Describe compatibility requirements for source generators.
+
+### Non-Goals
+
+- Define a process for components not in shared frameworks to contribute source generators.
+- Define any changes to the source generator architecture.
+- Define any implementation details of source generators or source generator tests.
+- Define a process around internal or non-shipping source generators
+
+## Stakeholders and Reviewers
+
+- ASP.NET - also building shared framework source generators.
+- .NET Libraries team - building multiple shared framework source generators.
+- Mono team - interested in providing source generators.
+- Roslyn team - responsible for source generator infrastructure.
+- .NET SDK team - responsible for resolving shared frameworks and doing conflict resolution with content.
+- NuGet team - FYI only, no changes requested
+
+## Design
+
+### Framework provided source generators
+
+Framework provided source generators should have a mechanism to be exposed and versioned along side the framework and library which they extend.  These source generators will be passed to the compiler when that framework is targeted.  These source generators will not ne used when the framework is not targeted.  For example, ASP.NET specific source generators will not be used when ASP.NET is not referenced.  The version of the analyzers specific to the TargetFramework version will be used.  For example, when targeting .NET 6.0 the analyzers specific to net6.0 will be used and not those for .NET 7.0, even when using the SDK that ships with .NET 7.0.
```

Comment:
How are the source generators going to interact with optional workloads?  I'm concern about ref pack dependency which is not known for some of the optional workloads from a runtime perspective.

Reply:
Optional workloads can provide their own targeting packs, so they could also provide source generators to go along with them.  They would need to follow the same restrictions about the dependencies they can take.  Is that what your concern is?

Reply:
No, what I had in mind is that optional workloads could share some part of tooling (e.g. AOT toolchain) which might require source generators. Such code generators could be workload or runtime pack specific.

Reply:
Today runtimes don't leak into the compile phase.  I think if you start doing that then you really should have a different TFM, or we should be adding support for cross-compilation by runtime cc @terrajobst

Reply:
You can see a bit of this in the Q & A section titled 
> Why in the ref pack, why not runtime pack?